docassemble/AssemblyLine/sessions.py

```diff
@@ -518,21 +519,31 @@ def nice_interview_title(
         return word("Untitled interview")
 
 
+def pascal_to_zwspace(text: str) -> str:
+    """
+    Insert a zero-width space into words that are PascalCased to help
+    with word breaks on small viewports.
+    """
+    re_outer = re.compile(r"([^A-Z ])([A-Z])")
+    re_inner = re.compile(r"(?<!^)([A-Z])([^A-Z])")
+    return re_outer.sub(r"\1​\2", re_inner.sub(r"​\1\2", text))
```

Comment:
I'm not sure what I'm doing wrong, but I can't get this to work: https://replit.com/@plocket/scratchpad#main.py. Also, I don't see spaces in here, but strange red dots appeared in the replit editor. When I add actual spaces, it seems to work fine.

```suggestion
    re_outer = re.compile(r"([^A-Z ])([A-Z])")
    re_inner = re.compile(r"(?<!^)([A-Z])([^A-Z])")
    return re_outer.sub(r"\1 \2", re_inner.sub(r" \1\2", text))
```

Reply:
We don't want a real space. The strange red dot is a zero width (invisible) space that is only used to control where line breaks are inserted.